src/chainstate/stacks/db/mod.rs

```diff
@@ -894,87 +904,51 @@ impl StacksChainState {
+                let initial_vesting_schedules = get_schedules();
+                for schedule in initial_vesting_schedules {
+                    let value = Value::Tuple(
+                        TupleData::from_data(vec![
+                            (
+                                "recipient".into(),
+                                Value::Principal(schedule.address.into()),
+                            ),
+                            ("amount".into(), Value::UInt(schedule.amount.into())),
+                        ])
+                        .unwrap(),
+                    );
+                    match vesting_schedules_per_block.entry(schedule.block_height) {
+                        Entry::Occupied(schedules) => {
+                            schedules.into_mut().push(value);
+                        }
+                        Entry::Vacant(entry) => {
+                            let schedules = vec![value];
+                            entry.insert(schedules);
+                        }
+                    };
+                }
+                let lockup_contract_id = boot_code_id("lockup");
                 clarity_tx.connection().as_transaction(|clarity| {
-                    let initial_vesting_schedules = get_schedules();
-                    let mut unlocks_per_blocks: HashMap<u64, u32> = HashMap::new();
-                    let lockup_contract_id = boot_code_id("lockup");
-                    for schedule in initial_vesting_schedules {
-                        total_vesting_amount = total_vesting_amount
-                            .checked_add(schedule.amount as u128)
-                            .expect("FATAL: vesting STX overflow");
-                        vesting_schedule_count = vesting_schedule_count + 1;
-                        let value = unlocks_per_blocks.get(&schedule.block_height).unwrap_or(&0);
-                        let index = value + 1;
-                        unlocks_per_blocks.insert(schedule.block_height, index);
-
-                        clarity
-                            .with_clarity_db(|db| {
-                                let key = TupleData::from_data(vec![
-                                    (
-                                        "stx-height".into(),
-                                        Value::UInt(schedule.block_height.into()),
-                                    ),
-                                    ("index".into(), Value::UInt(index.into())),
-                                ])
-                                .unwrap();
-
-                                let value = TupleData::from_data(vec![
-                                    ("owner".into(), Value::Principal(schedule.address.into())),
-                                    ("metadata".into(), Value::buff_from_byte(0)),
-                                    ("unlock-ustx".into(), Value::UInt(schedule.amount.into())),
-                                ])
-                                .unwrap();
+                    clarity
```

Comment:
I feel weird about this, now that it's finally clear how `chainstate.txt` is getting used to instantiate the initial balances and token unlock schedules.

We're going through a lot of trouble to generate a `chainstate.txt` file, only to load it up via `get_schedules()` and import the lines into the Clarity VM as entries in a data map.  If the `lockup` data map is the final destination for this data, then why not just generate the file `lockup.clar` so that it has a `(begin ..)` code block that contains all of the Clarity code needed to populate the data map?  Then, there's no need for most of this code -- all this code would need to do is load up the code from the `lockup.clar` text file and instantiate it as a contract.

Reply:
I like the elegancy of your proposal, but I'm sceptic on the overhead that it would create - the vm would have to be able to ingest a 20 mb contract. It sounds like a bet we can take when everything else is secured?

Reply:
Yes, runtime is a major consideration.  Loading the 35kb PoX contract takes ~5 seconds on my laptop.  If this scales up, then a 20mb contract might take about 47 _minutes_ (and who knows how much memory).

Regardless, once we choose how we're going to ingest the data, we can't change it later -- it's consensus-critical since it affects the final root hash of the genesis block.  So let's strive to be happy with whatever decision we make -- we're stuck with it.

Do you have any performance numbers on how long it takes to load a production `chainstate.txt` into the data map?

Reply:
When I run this against a prod chainstate.txt, this vesting code takes ~2 seconds to complete on my machine. There are ~60k lockup entries. 

Loading the 300k liquid balances takes much longer, around 70 seconds (that code path is unrelated to this PR).